commands/01-mod.js

```diff
@@ -158,18 +158,37 @@ const Module = new Augur.Module()
           if (bannerHighRole.comparePositionTo(bannedHighRole) <= 0) {
             msg.reply(`you cannot ban ${u.escapeText(member.displayName)}!`);
             continue;
-          } else {
+          }
+          members.set(userId, member);
+        }
+        else {
+          members.set(userId, undefined);
+        }
+      }
+    }
+    if (members.size == 0) {
+      msg.reply("you need to tell me who to ban!").then(u.clean);
+      return;
+    }
+
+    let confirm = await u.confirm(msg, `Are you sure you want to ban the following?\n${members.keyArray().map(m => members.get(m) ? u.escapeText(members.get(m).displayName) : m).join("\n")}`);
+
+    if (confirm) {
+      let banCount = 0;
+      for (const [memberId, member] of members) {
+        try {
+          if (member) {
             const infraction = {
               discordId: member.id,
               description: reason,
               value: 30,
               mod: msg.author.id
             };
```

Comment:
Next few changes are to consolidate the process, whether or not the member is found.
```suggestion
            const infraction = {
              discordId: memberId,
              description: reason,
              value: 30,
              mod: msg.author.id
            };
```